File: frontend/src/api/conversations.ts
import { http } from './http'
import type { Conversation, Message } from '../types'

export function listConversations(): Promise<Conversation[]> {
  return http.get<Conversation[]>('/api/conversations')
}

export function createConversation(title?: string): Promise<Conversation> {
  return http.post<Conversation>('/api/conversations', title ? { title } : undefined)
}

export function getMessages(conversationId: string): Promise<Message[]> {
  return http.get<Message[]>(`/api/conversations/${conversationId}/messages`)
}

export function deleteConversation(conversationId: string): Promise<void> {
  return http.delete<void>(`/api/conversations/${conversationId}`)
}

File: app/api/conversations.py
"""会话管理 API：列表 / 新建 / 查看消息 / 删除。"""
import uuid

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import select
from sqlalchemy.orm import Session

from app.database import get_db
from app.models import Conversation, Message
from app.schemas import ConversationOut, CreateConversationRequest, MessageOut

router = APIRouter()


@router.get("", response_model=list[ConversationOut])
def list_conversations(db: Session = Depends(get_db)):
    stmt = select(Conversation).order_by(Conversation.updated_at.desc())
    return db.execute(stmt).scalars().all()


@router.post("", response_model=ConversationOut, status_code=201)
def create_conversation(
    req: CreateConversationRequest | None = None,
    db: Session = Depends(get_db),
):
    title = (req.title if req else None) or "新对话"
    conv = Conversation(title=title)
    db.add(conv)
    db.commit()
    db.refresh(conv)
    return conv


@router.get("/{conversation_id}/messages", response_model=list[MessageOut])
def get_messages(conversation_id: uuid.UUID, db: Session = Depends(get_db)):
    if db.get(Conversation, conversation_id) is None:
        raise HTTPException(status_code=404, detail="会话不存在")
    stmt = (
        select(Message)
        .where(Message.conversation_id == conversation_id)
        .order_by(Message.created_at.asc())
    )
    return db.execute(stmt).scalars().all()


@router.delete("/{conversation_id}", status_code=204)
def delete_conversation(conversation_id: uuid.UUID, db: Session = Depends(get_db)):
    conv = db.get(Conversation, conversation_id)
    if conv is None:
        raise HTTPException(status_code=404, detail="会话不存在")
    db.delete(conv)
    db.commit()
